File: app/src/main/java/com/example/carebear/adapters/ChatAdapter.java
package com.example.carebear.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.carebear.R;
import com.example.carebear.models.ChatMessage;

import java.util.ArrayList;
import java.util.List;

public class ChatAdapter extends RecyclerView.Adapter<ChatAdapter.MessageViewHolder> {

    private final List<ChatMessage> messages = new ArrayList<>();

    public void addMessage(ChatMessage message) {
        messages.add(message);
        notifyItemInserted(messages.size() - 1);
    }

    @NonNull
    @Override
    public MessageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        int layoutId = viewType == 1 ? R.layout.item_message : R.layout.item_message_received;
        View view = LayoutInflater.from(parent.getContext()).inflate(layoutId, parent, false);
        return new MessageViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MessageViewHolder holder, int position) {
        holder.bind(messages.get(position));
    }

    @Override
    public int getItemCount() {
        return messages.size();
    }

    @Override
    public int getItemViewType(int position) {
        return messages.get(position).isSent() ? 1 : 0;
    }

    static class MessageViewHolder extends RecyclerView.ViewHolder {
        private final TextView tvMessageContent;

        MessageViewHolder(View itemView) {
            super(itemView);
            tvMessageContent = itemView.findViewById(R.id.tv_message_content);
        }

        void bind(ChatMessage message) {
            tvMessageContent.setText(message.getText());
        }
    }
}

File: app/src/main/java/com/example/carebear/adapters/ChatAdapter.kt
package com.example.carebear.adapters

import android.content.Context
import android.os.Build
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.recyclerview.widget.RecyclerView
import com.example.carebear.R
import com.example.carebear.models.ChatMembership
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class ChatAdapter(
    private val context: Context,
    private val chatMemberships: List<ChatMembership>,
    private val onItemClick: (ChatMembership) -> Unit
) :
    RecyclerView.Adapter<ChatAdapter.ChatViewHolder>() {

    class ChatViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val textName: TextView = itemView.findViewById(R.id.text_name)
        val lastMessage: TextView = itemView.findViewById(R.id.last_message)
        val date: TextView = itemView.findViewById(R.id.text_date)
        val hour: TextView = itemView.findViewById(R.id.text_hour)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChatViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_chat, parent, false)
        return ChatViewHolder(view)
    }


    @RequiresApi(Build.VERSION_CODES.O)
    override fun onBindViewHolder(holder: ChatViewHolder, position: Int) { // TODO
        val chat = chatMemberships[position]

        holder.textName.text = chat.recipientName
//        holder.lastMessage.text = chat.lastMessage
//
//        val timestamp = Date(chat.timestamp)
//        val dateFormat = SimpleDateFormat("dd/MM/yyyy", Locale.getDefault())
//        holder.date.text = dateFormat.format(timestamp)
//        val minuteHourFormat = SimpleDateFormat("HH:mm", Locale.getDefault())
//        holder.hour.text = minuteHourFormat.format(timestamp)

        holder.itemView.setOnClickListener {
            onItemClick(chat) // Trigger the callback with the clicked item
        }
    }

    override fun getItemCount() = chatMemberships.size
}